### ISSUE TITLE
Recommended approach to running tests on build server?

### ISSUE THREAD

**Issue by justinhelgerson** [Microsoft/testfx#287]
## Description
I'm building a new build server for my .NET (primarily 4.6.2) applications with the goal of not needing to install a full copy of Visual Studio.  Previously, we were referencing the Visual Studio test assembly in our test projects and invoking `mstest.exe` during our build process to run the tests.  After doing some reading, it sounds like the Microsoft testing world is changing for the better.

I replaced the Visual Studio test assembly with the `MSTest.TestFramework` NuGet package.  However, now my questions are:

1. How do I invoke my tests?  It sounds like vstest.console.exe is needed, however that doesn't appear to come with the Visual Studio Build Tools download.  Am I forced into installing a full copy of Visual Studio?
2. The `MSTest.TestFramework` package makes a reference to the `MSTest.TestAdapter` package as being used to execute tests.  However, as far as I can tell, it doesn't provide a way to execute the tests?  What's the purpose of the `TestAdapter` package?

Thanks!

**Comment by AbhitejJohn** [comment]
That is correct. vstest.console is what you need. You could pull in the tooling package from [here](https://dotnet.myget.org/feed/vstest/package/nuget/Microsoft.TestPlatform) to run your tests independent of VS. That said, we are also working on an official packaging story here.
The `MSTest.TestAdapter` package is an extension to the console runner that actually executes tests. `MSTest.TestFramework` package contains all the framework constructs that test authors can attribute to their test code.

**Comment by hickford** [comment]
So [Build Tools for Visual Studio 2017](https://www.visualstudio.com/downloads/#build-tools-for-visual-studio-2017) doesn't include `vstest.console.exe` ?

Found it in https://www.nuget.org/packages/Microsoft.TestPlatform/

**Comment by justinhelgerson** [comment]
@hickford It's also included in https://www.visualstudio.com/downloads/#agents-for-visual-studio-2017 which is what I used for my build server.  Just FYI.

**Comment by hickford** [comment]
Thanks @justaniles, that's useful to know.

**Comment by anushaande** [comment]
Hi! This is Anusha. I am new to .net and Docker containers. We started a project to build CI to build, test and deploy the .net framework application. I want to containerize the application and ship it through the entire workflow of the CI pipeline. I was able to build the application in a container and deploy it to IIs container. I need a container to run unit tests in between.
To get a knowledge of this, I was going through different blogs. I found this article useful and installed Microsoft.TestPlatform package in my docker container. Could someone help me with next steps till i will be able to run unit tests successfully.
